File: Ch. 1/URLify.cpp
#include <iostream>
#include <string>

/* Replace all spaces in a string with "%20" */

int main() {
   std::string ret = "";
   std::string s;
   std::getline(std::cin, s);
   int length;
   std::cin >> length;
   for (int i = 0; i < length; i++) {
      if (isspace(s[i])) {
         ret = ret + "%20";
      }
      else {
         ret = ret + s[i];
      }
   }
   std::cout << ret << "\n";
   return 0;
}

File: Ch. 1/URLify.py
# Replace all spaces in a string with "%20"

# Loop through each character O(n)
def replaceSpace_1(s):
    return_str = ''

    for char in s:
        if char != ' ':
            return_str += char
        else:
            return_str += '%20'

    return return_str

test_str = input('Enter string: ')
print(replaceSpace_1(test_str))

